test/linalg.jl

```diff
@@ -245,6 +245,21 @@ end
             end
         end
     end  # SVD
+    @testset "Left multiplication" begin
+        N1 = 20
+        N2 = 8
+        N3 = 5
+        N4 = N1 + N3 - N2
+        A = BlockDiagonal([rand(rng, N1, N1), rand(rng, N2, N2)])
+        B = BlockDiagonal([rand(rng, N1, N2), rand(rng, N3, N4)])
+        x = rand(rng, N1 + N2)
+        y = rand(rng, N2 + N4)
+
+        @testset "Lower triangular" begin
```

Comment:
Do these tests fail on master, or are they just slow? Perhaps we could test for performance as well, see NamedDims.jl which does a few allocation tests, maybe some speed tests as well.

(I'm thinking higher level test, i.e. sampling)

Reply:
The test pass on master, just allocate more. So if I add a test for the case in the #116 (and add `Distributions` to the test deps)?

Reply:
I have added an allocation test just on `lmul!`. I suggest we leave higher level testing to a separate PR and plan it together with a benchmark suite.